### ISSUE TITLE
DeprecationWarning: The epoch parameter in `scheduler.step()` was not necessary

### ISSUE THREAD

**Issue by bratao** [allenai/allennlp#3922]
**Describe the bug**
The Scheduler wrapper in Allennlp 

lib/python3.7/site-packages/torch/optim/lr_scheduler.py:591: DeprecationWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.

**To Reproduce**
Use  an scheduler
```
scheduler = ReduceOnPlateauLearningRateScheduler(
            optimizer, factor=0.5, patience=2, mode="min", verbose=True
        )
```

**System (please complete the following information):**
 - OS: Debian 10
 - Python version: 3.7.5
 - AllenNLP version: pip install allennlp==0.9.1.dev20200302
 - PyTorch version: 1.4.0

**Comment by kernelmachine** [comment]
Thanks for raising this! @dirkgr , can you take a look?

**Comment by dirkgr** [comment]
I think we should fix this, but it's not more important than the items we have in the 1.0 milestone right now.
